### ISSUE TITLE
Composer adds extra / to PSR-4 autoload mapping

### ISSUE THREAD

**Issue by Giblin** [composer/composer#4098]
This may be related to issue #3024 but I'm some what new to composer and not 100% sure it's the same issues as I'm not on a network mount and I'm not getting `../`

I have a VM of Ubuntu 14.04 server running in VMware player on my local machine with a full LAMP stack.  I installed composer and copied it to the `/bin/local/`  like the documentation states to make composer global. This works great, but I'm wondering if this may be the underlining cause to the issue I'm having.  In my test project i have the following in my composer.json in the root of my project.

```
"autoload" : {
	"psr-4" : {
		"VendorName\\NewProject\\" : "src"	
	}
}
```
I then run `composer dump-autoload -o` in the root of my project. (I have also tried it with out the -o).  Then when I try to run my unit test, php cant find the classes in src.  If i go and look at `vendor/composer/autoload_psr4.php` is see the following.

```
<?php

// autoload_psr4.php @generated by Composer

$vendorDir = dirname(dirname(__FILE__));
$baseDir = dirname($vendorDir);

return array(
    'VendorName\\NewProject\\' => array($baseDir . '/src'),
);
```
There is a `/ `in front of my src directory.  This makes php think the file is at the root of the file structure not the root of my project.  If I manually remove the / and save the file everything works great.

Is this because I have composer installed globally? Is there a way to tell it not to do this?

**Comment by stof** [comment]
if you remove the / here, it breaks everything because ``$vendorDir`` does not have a trailing slash

**Comment by Giblin** [comment]
No it dose not break everything.  If I leave it in it breaks everything.  When i pull it out everything works like it should Phing, PHPunit all run flawlessly.  All my test run and work exactly as I would expect.  If i leave it in PHPunit brakes and cant find the class files in the src directory because it dose not exists at the file system root.

**Comment by stof** [comment]
I mean it will breaks everything for any other user of Composer where the vendor dir is detected properly.

Can you give more details about where the project root is and how the composer.json file looks like ?

**Comment by Giblin** [comment]
Sure

Project Directory is in `/var/www/vendorname/newproject`
Structure looks like this
-src
    heloworld.php
-tests
    heloworldtest.php
-vendor
   -bin
   -composer
   -phpunit
   .....
build.xml
composer.json
composer.lock

the composer.josn file looks like this.
```
{
	"require-dev" : {
        "phpunit/phpunit": "4.8.*@dev"
    },
	"autoload" : {
		"psr-4" : {
			"VendorName\\NewProject\\" : "src"	
		}
	}
}
```

**Comment by stof** [comment]
This looks really weird. Can you do a var_dump of ``$basedir`` in your ``autoload_psr4.php`` file ? I don't see why it would be empty, but this seems the only way for the path to be considered as being at the filesystem root.

**Comment by Giblin** [comment]
I agree its weird

Ok so I did a dump and it returned `/var/www/vendorname/newproject`

I ran my phpUnit again and they where working like before. So for giggles I put the `/` back in and tried again.....and it works, both ways, with and with out the `/` in the `autoload_psr4.php`

I ran `dump-auotload -o` again and let composer rest the file to make sure I had not forgot to remove my personal edits, ran my test and its working!

I'm sorry to have wasted your time.  I was doing most of this late last night and must have messed something up that got reset in today's boot of the system. 

Thank you for your help and patience with this.
